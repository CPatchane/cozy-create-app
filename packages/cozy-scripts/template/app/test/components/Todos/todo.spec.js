'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// we import the not connected component version for testing
import { Todo } from 'components/Todos/Todo'

// mock data for testing
import mockTodos from './_mocksTodos.json'

Enzyme.configure({ adapter: new Adapter() })

// async mock function
const mockUpdateTodo = jest.fn(() => Promise.resolve())
const todo = mockTodos[0]

describe('Todo component:', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <Todo updateTodo={mockUpdateTodo} todo={todo} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should handle isWorking correctly (display spinner)', () => {
    const component = shallow(<Todo updateTodo={mockUpdateTodo} todo={todo} />)
    expect(component.state().isWorking).toBe(false)
    component.setState({ isWorking: true })
    expect(component).toMatchSnapshot()
  })
})
