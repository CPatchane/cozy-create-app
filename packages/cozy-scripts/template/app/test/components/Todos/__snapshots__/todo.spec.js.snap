// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component: should be rendered correctly 1`] = `
<span
  onDoubleClick={[Function]}
>
  Todo 1
</span>
`;

exports[`Todo component: should handle isWorking correctly (display spinner) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    todo={
      Object {
        "_id": 1234,
        "name": "Todo 1",
      }
    }
    updateTodo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Todo 1",
      "onDoubleClick": [Function],
    },
    "ref": null,
    "rendered": "Todo 1",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Todo 1",
        "onDoubleClick": [Function],
      },
      "ref": null,
      "rendered": "Todo 1",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
